static/src/javascripts/projects/common/modules/commercial/fiv-banner.js

```diff
@@ -0,0 +1,132 @@
+const maxArticles = 3;
+const fivFirstSeenAtKey = 'fivFirstSeenAt';
+
+const getVisitCount = (): number => local.get('gu.alreadyVisited') || 0;
+
+const defaultEngagementBannerParams = () => {
+    const linkUrl = "https://www.theguardian.com/"; // FIXME
+
+    return {
+        campaignCode: 'gdnwb_copts_fiv_banner',
+        pageviewId: config.get('ophan.pageViewId', 'not_found'),
+        products: ['CONTRIBUTION', 'RECURRING_CONTRIBUTION'],
+        linkUrl: supportContributeURL,
+        hideBanner: hideBanner,
+    };
+};
+
+const hideBanner = () => {
+    userPrefs.set(fivFirstSeenAtKey, maxArticles * -1);
+};
+
+const showBanner = (params: EngagementBannerParams): void => {
+
+    const linkUrl = addTrackingCodesToUrl({
+        base: params.linkUrl,
+        componentType: 'ACQUISITIONS_ENGAGEMENT_BANNER',
+        componentId: params.campaignCode,
+        campaignCode: params.campaignCode,
+    });
+
+    const location = getGeoLocation();
+    const renderedBanner: string = acquisitionsBannerFivTemplate(location, linkUrl);
+    const messageShown = new Message(messageCode, {
+        siteMessageLinkName: 'membership message',
+        siteMessageCloseBtn: 'hide',
+        siteMessageComponentName: params.campaignCode,
+        trackDisplay: true,
+        cssModifierClass: 'fiv-banner',
+        customJs() {
+            bean.on(
+                document,
+                'click',
+                '.js-fiv-banner-close-button',
+                () => {
+                    this.hide();
+                    hideBanner();
+                }
+            );
+        },
+    }).show(renderedBanner);
+
+    if (messageShown) {
+        ['INSERT', 'VIEW'].forEach(action => {
+            submitComponentEvent({
+                component: {
+                    componentType: 'ACQUISITIONS_ENGAGEMENT_BANNER',
+                    products: params.products,
+                    campaignCode: params.campaignCode,
+                    id: params.campaignCode,
+                },
+                action,
+            });
+        });
+
+        mediator.emit('membership-message:display');
+    }
+};
+
+const show = (): Promise<boolean> => {
+    const params = defaultEngagementBannerParams();
+
+    if (params) {
+        showBanner(params);
+        return Promise.resolve(true);
+    }
+    return Promise.resolve(false);
+};
+
+const canShow = (): Promise<boolean> => {
+
+    // always show if they put #fiv in the url hash
+    if (window.location.hash.match(/[#&]fiv(&.*)?$/)) {
+        console.log("fiv force ON");
+        return Promise.resolve(true);
+    }
+
+    if (!config.get('switches.fivBanner')/*TODO create the switch*/ || isBlocked()/*TODO CHECK THIS*/) {
+        return Promise.resolve(false);
+    }
+
+    if (
+        shouldShowReaderRevenue()
+    ) {
+        const fivFirstSeenAt = userPrefs.get(
+            fivFirstSeenAtKey
+        );
+        const visitCount = getVisitCount();
+
+        if (!fivFirstSeenAt) {
```

Comment:
Could we invert the condition here? I think it might be easier to reason about:
```js
if (fivFirstSeenAt) {
```
Also we don't need an `else` block if we return in the `if`